VideoLocker/src/main/java/org/edx/mobile/social/google/GoogleProvider.java

```diff
@@ -0,0 +1,90 @@
+package org.edx.mobile.social.google;
+
+import android.app.Activity;
+import android.content.Context;
+
+import org.edx.mobile.model.api.CourseEntry;
+import org.edx.mobile.social.SocialGroup;
+import org.edx.mobile.social.SocialMember;
+import org.edx.mobile.social.SocialProvider;
+
+import java.util.List;
+
+/**
+ * To make an consistent access interface to google as facebook used.
+ *
+ * Created by hanning on 3/12/15.
+ */
+public class GoogleProvider implements SocialProvider{
+
+    private GoogleOauth2 google;
+
+    public GoogleProvider(GoogleOauth2 google){
+        this.google = google;
+    }
+
+    @Override
+    public boolean isLoggedIn() {
```

Comment:
It might be worth breaking SocialProvider into multiple parts, one about login and one with all the other stuff. That way we can naturally support multiple login sources, but only one sharing source?